server/src/main/java/io/crate/role/DropRoleTask.java

```diff
@@ -74,19 +79,18 @@ static boolean dropRole(Metadata.Builder mdBuilder, String roleNameToDrop) {
 
         UsersPrivilegesMetadata oldUserPrivilegesMetadata = (UsersPrivilegesMetadata) mdBuilder.getCustom(UsersPrivilegesMetadata.TYPE);
         RolesMetadata newMetadata = RolesMetadata.of(mdBuilder, oldUsersMetadata, oldUserPrivilegesMetadata, oldRolesMetadata);
-        validateHasChildren(newMetadata.roles().values(), roleNameToDrop);
+        ensureHasNoDependencies(newMetadata.roles().values(), roleNameToDrop);
         var role = newMetadata.remove(roleNameToDrop);
         if (role == null && newMetadata.equals(oldRolesMetadata)) {
```

Comment:
`newMetadata.equals(oldRolesMetadata)` should not be needed I believe, the assert below should be enough.

Reply:
This seems weird, indeed, please let me do some tests before we remove this though, I'll take care of it myself, thx!

Reply:
:+1:

Reply:
Looks like it is from https://github.com/crate/crate/pull/15261, and I partially broke the logic.